src/charm.py

```diff
@@ -467,6 +475,42 @@ def _has_required_fields(self, conf_or_rel: _ConfigOrRelation):
         field_names = [f'_{f.replace("-", "_")}' for f in REQUIRED_INGRESS_RELATION_FIELDS]
         return all(getattr(conf_or_rel, f) for f in field_names)
 
+    def _delete_unused_services(self, current_svc_names: List[str]):
```

Comment:
Was the mypy configuration added? I can't spot the change in this PR?

Reply:
Sorry, I just added the "-> None". Next commit will have mypy.

Reply:
BTW I added a lot of "type: ignore"/"disable"s but without comment because would be repetitive. I'm not sure if this is the best approach.
There are many cases like:
```
tests/integration/conftest.py:40: error: Item "None" of "Optional[Model]" has no attribute "deploy"  [union-attr]
```

Reply:
So for those just put an `assert app.model` or whatever at the start of the test and they will go away. Mypy is telling you that `ops` doesn't guarantee the model is defined - therefore we should check that it is if we are going to use it

Reply:
Thanks!